src/main/java/com/aws/greengrass/lifecyclemanager/KernelCommandLine.java

```diff
@@ -64,6 +64,7 @@ public class KernelCommandLine {
     private static final String deploymentsPathName = "~root/deployments";
     private static final String cliIpcInfoPathName = "~root/cli_ipc_info";
     private static final String binPathName = "~root/bin";
+    private static final String validUserCharExpr = "^[a-zA-Z0-9._]+[a-zA-Z0-9._-]*$";
```

Comment:
Would this false-warn if someone running this on linux supplies a valid `user:group` string? The requirements are different between linux and windows I thought

Reply:
You're right - it doesn't look like the group is parsed from the user so it would give a false warning. I'll move this inside the windows conditional.